### PR TITLE
[TASK] Switch to the core testing framework

### PR DESCRIPTION
**Pull request by oliverklee**
The TYPO3 core testing framework has more person-power for maintenance
behind it compared to the nimut testing framework. So we should use that.

### CODE REVIEW

**Comment by DanielSiepmann** [comment]
I don't know current state, but https://github.com/sabbelasichon/typo3-rector/blob/main/src/Set/Extension/NimutTestingFrameworkSetList.php might be helpful. We used the set during a project update from V8 to V10 to also switch the framework, and it did the job.

**Comment by oliverklee** [comment]
@DanielSiepmann Thanks!

File: Tests/Unit/Controller/TeaControllerTest.php [hunk 1]
```diff
@@ -36,14 +36,17 @@ class TeaControllerTest extends UnitTestCase
...
+        $this->subject = $this->getAccessibleMock(TeaController::class, ['redirect', 'forward']);
 
         $this->viewProphecy = $this->prophesize(TemplateView::class);
         $view = $this->viewProphecy->reveal();
-        $this->inject($this->subject, 'view', $view);
+        $this->subject->_set('view', $view);
 
         $this->teaRepositoryProphecy = $this->prophesize(TeaRepository::class);
-        /** @var TeaRepository&ProphecySubjectInterface $teaRepository */
+        /** @var TeaRepository|ProphecySubjectInterface $teaRepository */
```
Comment (oliverklee) [inline Tests/Unit/Controller/TeaControllerTest.php:49]: We need to undo this change. (PHP-CS-Fixer complaining about this is a bug avoided by #371.)

**Comment by lukaszuznanski** [comment]
I got totally different errors locally in functional tests.

**Comment by oliverklee** [comment]
This PR needs rebasing.

File: Tests/Functional/Domain/Repository/Product/TeaRepositoryTest.php [hunk 2]
```diff
@@ -122,11 +122,11 @@ public function addAndPersistAllCreatesNewRecord(): void
...
 
-        $databaseRow = $this->getDatabaseConnection()->selectSingleRow(
-            '*',
-            'tx_tea_domain_model_product_tea',
-            'uid = ' . $model->getUid()
-        );
+        $connection = $this->getConnectionPool()
+            ->getConnectionForTable('tx_tea_domain_model_product_tea');
+        $databaseRow = $connection->select(['*'], 'tx_tea_domain_model_product_tea', ['uid' => $model->getUid()])
+            ->fetchAssociative();
```
Comment (oliverklee) [inline Tests/Functional/Domain/Repository/Product/TeaRepositoryTest.php:128]: To be compatible with TYPO3 10LTS with lowest dependencies, we need to use the deprecated method here (maybe with a comment).

Comment (oliverklee) [reply Tests/Functional/Domain/Repository/Product/TeaRepositoryTest.php:128]: These are the requirements in the `composer.json` of `typo3/cms-core`:

- V10: `"doctrine/dbal": "~2.10.0 || ~2.11.2 || ~2.13.1",`
- V11: `"doctrine/dbal": "^2.13.5",`
- main: `"doctrine/dbal": "^3.2",`

**Review by lukaszuznanski — APPROVED** [review]
